drivers_linux.go

```diff
@@ -1,25 +1,66 @@
 package libnetwork
 
 import (
+	"strings"
+
 	"github.com/docker/libnetwork/driverapi"
 	"github.com/docker/libnetwork/drivers/bridge"
 	"github.com/docker/libnetwork/drivers/host"
 	"github.com/docker/libnetwork/drivers/null"
 	o "github.com/docker/libnetwork/drivers/overlay"
 	"github.com/docker/libnetwork/drivers/remote"
+	"github.com/docker/libnetwork/netlabel"
 )
 
-func initDrivers(dc driverapi.DriverCallback) error {
-	for _, fn := range [](func(driverapi.DriverCallback) error){
-		bridge.Init,
-		host.Init,
-		null.Init,
-		remote.Init,
-		o.Init,
-	} {
-		if err := fn(dc); err != nil {
+func initDrivers(c *controller) error {
+	initializers := []struct {
+		fn    func(driverapi.DriverCallback, map[string]interface{}) error
+		ntype string
+	}{
+		{bridge.Init, "bridge"},
+		{host.Init, "host"},
+		{null.Init, "null"},
+		{remote.Init, "remote"},
+		{o.Init, "overlay"},
+	}
+
+	for _, i := range initializers {
+		if err := i.fn(c, makeDriverConfig(c, i.ntype)); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
+
+func makeDriverConfig(c *controller, ntype string) map[string]interface{} {
+	if c.cfg == nil {
+		return nil
+	}
+
+	config := make(map[string]interface{})
+
+	if c.validateDatastoreConfig() {
+		config[netlabel.KVProvider] = c.cfg.Datastore.Client.Provider
+		config[netlabel.KVProviderURL] = c.cfg.Datastore.Client.Address
+	}
+
+	for _, label := range c.cfg.Daemon.Labels {
+		if !strings.HasPrefix(label, netlabel.DriverPrefix+"."+ntype) {
```

Comment:
To be more precise, can you change `strings.HasPrefix(label` to `strings.HasPrefix(netlabel.Key(label)`...
Though the current code works by the virtue of how the netlabel is defined. But still for technical accuracy,
I think ading the netlabel.Key() makes it more easily readable. WDYT ?